src/packages/storage-provider/src/decorators/media-field.ts

```diff
@@ -66,10 +66,10 @@ export class GraphweaverMedia {
 	static serialize = ({ value }: { value: unknown }) => {
 		if (value === null) return null;
 		if (isMedia(value)) {
-			return JSON.stringify({
+			return {
```

Comment:
I'm pretty certain this change is needed here. The Postgres driver is accepting the object, and it gets persisted in a format that allows key based querying from the jsonb column.

There is a potential matching change in the deserialise below, but I have left it there for backwards compatibility. It tries a json parse of the untyped incoming value is detected as a string.

Reply:
I suppose this is a breaking change, as existing users would now see differently formatted results in their underlying JSON and JSONB columns, although the deserialise will transparently handle either.

But an independent service reading these columns could potentially be broken by the change

Reply:
If we were persisting a string before instead of a JSON object in the JSONB column then that was 100% a bug and we should fix it.

The only other thought I have is we should give them a script they can run to update all the JSONB strings to actual JSONB values in place so they don't have to mutate them all through the backend to get them over to the originally intended format.

Also sounds like we should just remove the serialize / de-serialize from this entity functionality altogether eventually.

I'll note this in the release notes, thanks for bringing it to my attention!

Reply:
I've had a think about all the different ways we could 'script' that migration. It seems unlikely that a user would want to run the graphweaver CLI against their production app. The best I can come up with is to offer a new action button in the admin UI 'migrate data'. This could pop open a modal, and you could choose:
- column to migrate
- source data type
- target data type

What do you think? I can't think of a more 'kitchen sink included' style solution. I think anyone with real production data is probably running proper schema migration steps anyway, which is probably a better format for the fix.

![image](https://github.com/user-attachments/assets/ed0cefcd-eb7c-48a4-ac1a-40bc797246dd)

Reply:
I just mean an example SQL script that we can pop in the release notes, sorry, should have been more clear!

Reply:
OK I think this is what we're after, I think the PR is good to go now:
```
UPDATE submission
SET image = 
  CASE 
    WHEN jsonb_typeof(image) = 'string' THEN (image#>>'{}')::jsonb
    ELSE image
  END
WHERE jsonb_typeof(image) = 'string';
```